### ISSUE TITLE
Multiple GPUs

### ISSUE THREAD

**Issue by grungert** [RasaHQ/rasa#8369]
**Rasa version**: 2.3.0 

**Python version**: 3.8.5 

**Operating system** (windows, osx, ...): Ubuntu 18.04 

**Issue**:
```

Hi,

We want to train our model on multiple GPUs, so we changed the code and added a few lines to use the distribution strategy. The model was using GPUs in parallel until the training of the DIETClassifier started and the following error occurred: 

"ValueError: SyncOnReadVariable does not support `assign_add` in cross-replica context when aggregation is set to `tf.VariableAggregation.SUM`". 

We found out that this error was solved in Tensorflow 2.4. However, according to the available Rasa dependencies, it is possible to use the Tensorflow version up to 2.3. Do you have any suggestions on how to use multiple GPUs in another way? Do you already support it?

```

**Error (including full traceback)**:
```
ValueError: SyncOnReadVariable does not support `assign_add` in cross-replica context when aggregation is set to `tf.VariableAggregation.SUM`
``` 

**Command or request that led to error**:
```
rasa train nlu
```

**Comment by sara-tagger** [comment]
Thanks for the issue, @joejuzl will get back to you about it soon!

###### You may find help in the [docs](https://rasa.com/docs/) and the [forum,](https://forum.rasa.com/) too 🤗

**Comment by ashuamit8** [comment]
did u found any solution ?? or is there any approach to use multi Gpu context with rasa.
kindly help with required examples.
Thanks

**Comment by joejuzl** [comment]
Hi @grungert,

For more information regarding tuning performance see: https://rasa.com/docs/rasa/tuning-your-model#configuring-tensorflow.
If you need more information please start a thread on the forum: https://forum.rasa.com/

W.r.t to the Rasa tensorflow version, a bug fix in which has not been backported to 2.4 is preventing us from upgrading. 
See https://github.com/RasaHQ/rasa/issues/7619 for more details.

**Comment by grungert** [comment]
UPDATE!

So far, we have tested several combinations of Tensorflow and Rasa versions and concluded that Rasa is still not available for multi-GPU training. More precisely, these versions are tested: 
- Rasa 2.3.0 and Tensorflow 2.3.1,
- Rasa 2.3.0 and Tensorflow 2.4.0,
- Rasa 2.4.3 and Tensorflow 2.4.0,
- Rasa 2.4.3 and TensorFlow 2.5.0-rc1,
- Rasa 2.3.0 and Tensorflow 2.5.0-rc1.

All released Rasa versions are compatible with maximum Tensorflow 2.3.1, while for multi-GPU training Tensorflow 2.4.0, or higher, must be used.

**Comment by lsyin** [comment]
> UPDATE!
> 
> So far, we have tested several combinations of Tensorflow and Rasa versions and concluded that Rasa is still not available for multi-GPU training. More precisely, these versions are tested:
> 
> * Rasa 2.3.0 and Tensorflow 2.3.1,
> * Rasa 2.3.0 and Tensorflow 2.4.0,
> * Rasa 2.4.3 and Tensorflow 2.4.0,
> * Rasa 2.4.3 and TensorFlow 2.5.0-rc1,
> * Rasa 2.3.0 and Tensorflow 2.5.0-rc1.
> 
> All released Rasa versions are compatible with maximum Tensorflow 2.3.1, while for multi-GPU training Tensorflow 2.4.0, or higher, must be used.

Great job! So we cannot train diet model on multi gpus ?

**Comment by dengxingwang** [comment]
> **Rasa version**: 2.3.0
> 
> **Python version**: 3.8.5
> 
> **Operating system** (windows, osx, ...): Ubuntu 18.04
> 
> **Issue**:
> 
> ```
> 
> Hi,
> 
> We want to train our model on multiple GPUs, so we changed the code and added a few lines to use the distribution strategy. The model was using GPUs in parallel until the training of the DIETClassifier started and the following error occurred: 
> 
> "ValueError: SyncOnReadVariable does not support `assign_add` in cross-replica context when aggregation is set to `tf.VariableAggregation.SUM`". 
> 
> We found out that this error was solved in Tensorflow 2.4. However, according to the available Rasa dependencies, it is possible to use the Tensorflow version up to 2.3. Do you have any suggestions on how to use multiple GPUs in another way? Do you already support it?
> ```
> 
> **Error (including full traceback)**:
> 
> ```
> ValueError: SyncOnReadVariable does not support `assign_add` in cross-replica context when aggregation is set to `tf.VariableAggregation.SUM`
> ```
> 
> **Command or request that led to error**:
> 
> ```
> rasa train nlu
> ```

Could you pls share your code on how to use multiple GPUs?

**Comment by Barmin93** [comment]
Why was this issue closed, when there is no solution to this problem?
